packages/rs-dapi-client/src/lib.rs

```diff
@@ -0,0 +1,194 @@
+//! This crate provides [DapiClient] --- client for a decentralized API for Dash.
+//!
+//! # Examples
+//! ```no_run
+//! # use rs_dapi_client::{Settings, platform::GetIdentity, AddressList, DapiClient, DapiError};
+//! # let _ = async {
+//! let mut client = DapiClient::new(AddressList::new(), Settings::default());
+//! let request = GetIdentity { id: b"0".to_vec() };
+//! let response = client.execute(request, Settings::default()).await?;
+//!
+//! # Ok::<(), DapiError<_, _>>(())
+//! # };
+//! ```
+
+#![deny(missing_docs)]
+
+mod address_list;
+pub mod core;
+pub mod platform;
+mod settings;
+mod transport;
+
+use std::fmt;
+
+use backon::{ExponentialBuilder, Retryable};
+use tracing::Instrument;
+
+pub use address_list::AddressList;
+pub use settings::Settings;
+use transport::{TransportClient, TransportRequest};
+
+/// DAPI request.
+/// Since each of DAPI requests goes through the same execution process is was generalized
+/// with this trait to run requests the same way.
+pub trait DapiRequest: fmt::Debug {
+    /// Response type for the request.
+    type DapiResponse;
+
+    /// Settings that will override [DapiClient]'s ones each time the request is executed.
+    const SETTINGS_OVERRIDES: Settings;
+
+    /// Error that may happen during conversion from transport-specific response to the
+    /// DAPI response.
+    type Error: std::error::Error;
+
+    /// 1 to 1 mapping from the DAPI request to a type that represents a way for the data
+    /// to be fetched.
+    type TransportRequest: TransportRequest;
+
+    /// Get the transport layer request.
+    fn to_transport_request(&self) -> Self::TransportRequest;
+
+    /// Attempts to build DAPI response specific to this DAPI request from transport layer data.
+    fn try_from_transport_response(
+        transport_response: <Self::TransportRequest as TransportRequest>::Response,
+    ) -> Result<Self::DapiResponse, Self::Error>;
+}
+
+/// Access point to DAPI.
+#[derive(Debug)]
+pub struct DapiClient {
+    address_list: AddressList,
```

Comment:
Yes, it's intentionally out of scope of this PR and should be done separately since naive address list is at least something we can use on first iteration